config/src/tls_config.rs

```diff
@@ -0,0 +1,109 @@
+use crate::ConfigDescription;
+use crate::ConfigOption;
+use crate::Configuration;
+
+/// The settings defined in this struct are common to all components that employ SSL
+/// for transport layer security and potentially also authentication.
+/// Not all settings are always needed, in a scenario without client authentication
+/// no keystore is necessary for example.
+
+pub struct TlsConfig {}
+
+impl TlsConfig {
+    /// A setting to provide the path to a file which will be used as keystore
+    pub const KEYSTORE_LOCATION: ConfigOption = ConfigOption {
+        name: "tls-keystore-location",
+        default: "",
+        required: false,
+        takes_argument: true,
+        help: "The location of the keystore to use when connecting to the orchestrator, keystore \
+        should be in PKCS12 format.",
+        documentation: "Specify a file in PKCS12 format that should be used to obtain keys\
+        used for encryption.\
+        The keystore can contain additional keys beside the needed one, in that case the first \
+        suitable key that is found will be used.",
+    };
+
+    /// A setting to provide the password to be used to open the keystore that
+    /// was provided
+    pub const KEYSTORE_PASSWORD: ConfigOption = ConfigOption {
+        name: "tls-keystore-password",
+        default: "",
+        required: false,
+        takes_argument: true,
+        help: "The password that is necessary to access the keystore, if one is required.",
+        documentation: "The password that is necessary to access the keystore, if one is required.",
+    };
+
+    pub const TRUSTSTORE_LOCATION: ConfigOption = ConfigOption {
+        name: "tls-truststore-location",
+        default: "",
+        required: false,
+        takes_argument: true,
+        help: "The location of the truststore to use when connecting to the orchestrator.",
+        documentation:
+            "Specify a file in PKCS12 format that should be used to check if certificates\
+        are signed by a trusted authority. \
+        Any certificate that was signed with the private key belonging to one of the public keys\
+        in this truststore will be accepted as a valid certificte by this client.",
+    };
+
+    pub const TRUSTSTORE_PASSWORD: ConfigOption = ConfigOption {
```

Comment:
Have you checked how TLS works in Rust? Are things called Key- & Truststore, do both concepts exist etc.?